conda_devenv/devenv.py

```diff
@@ -387,7 +387,18 @@ def truncate_history_file(env_directory):
 
 
 def __call_conda_env_update(args, output_filename):
+    import psutil
+
     env_manager = sys.argv[0]
+    for proc in psutil.Process().parents():
```

Comment:
Hmm I would rather avoid this kind of hack if possible.

@marcelotrevisani do you have any idea on how to know if we are being executed via `mamba` or `conda`?

Reply:
the problem is that, everything is a process here right? you are not using the conda API 
but shouldn't the ``sys.argv[0]`` have this information? if it is calling devenv from mamba or conda
because it should be mamba or conda as the first element

Reply:
it might be a path to the mamba or conda executable as well, but you can parse it
or am I missing something here?

Reply:
in case that is a path you might do something like
```
from pathlib import Path

Path(sys.argv[0]).stem
```
that should return mamba or conda

Reply:
```

Reply:
It is important to know that to implement those conda sub commands (`conda devenv`) we create an entry point called `conda-devenv` and conda relays the execution to it.

Reply:
it is the same for mamba/boa with mambabuild

Reply:
What? The "it relies on entry points" on "prusse is doing something wrong"?

Reply:
We could add a flag to devenv to select and env manager. I don't like this idea. A flag to select the env manager but fallback to the one the user did already select in the original command line is good, but as the sole way the use mamba no.

(I did try this, it dos not work 🥲 ) Different entry points will be a way to achieve that, in the recipe:
```
build:
  entry_points:
    - conda-devenv = conda_devenv.devenv:main
    - mamba-devenv = conda_devenv.devenv:main_with_mamba_as_default_env_manager
```

Reply:
+1 for that

Reply:
OK.
I will add the flag but default to the value obtained by that "process crawl stuff" obtained value.

Reply:
Wait, no.

I think we should just add the alternate entry point like you proposed, and let the user be explicit about what to call. No flags, and no automatic detection.

Reply:
> Or you are suggesting the user call mamba-devenv explicitly?

Exactly!

If we can't detect this simply and reliably, better to let the user be explicit. No big deal.